src/dec128.rs

```diff
@@ -1033,4 +1107,42 @@ mod tests {
         assert_eq!(d128!(1.1), d128!(1.1).min(d128!(2.2)));
         assert_eq!(d128!(1.1), d128!(1.1).min(&d128!(2.2)));
     }
+
+    fn compare_u8_arrs(left: &[u8], right: &[u8]) {
+        assert_eq!(left.len(), right.len());
+
+        for i in 0..left.len() {
+            if left[i] != right[i] {
+                panic!("compare_u8_arrs inequality: left {:?} right {:?}", left, right);
+            }
+        }
+    }
+
+    #[test]
+    fn double_dabble_test() {
+        let mut arr = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0];
+        double_dabble(18446744073709551615, &mut arr);
+        compare_u8_arrs(&[1,8,4,4,6,7,4,4,0,7,3,7,0,9,5,5,1,6,1,5], &arr);
+    }
+
+    #[test]
+    fn from_i64() {
+        assert_eq!(d128::from_str(&::std::i64::MAX.to_string()).unwrap(), d128::from(::std::i64::MAX));
+        assert_eq!(d128::from_str(&(::std::i64::MIN + 1).to_string()).unwrap(), d128::from(::std::i64::MIN + 1));
+
+        //sanity
+        assert_eq!(d128::from(123i32), d128::from(123i64));
```

Comment:
Haha, I was not  even sure myself that this was working, so added these. 
Removed now!